akka-stream/src/main/scala/akka/stream/scaladsl/One2OneBidiFlow.scala

```diff
@@ -77,10 +75,7 @@ object One2OneBidiFlow {
 
       setHandler(outOut, new OutHandler {
         override def onPull(): Unit = pull(outIn)
-        override def onDownstreamFinish(): Unit = {
-          cancel(outIn)
-          cancel(inIn) // short-cut to speed up cleanup of upstream
-        }
+        override def onDownstreamFinish(): Unit = cancel(outIn)
```

Comment:
> Yes. I always keep forgetting that there can only every be one pull and one push pending.

`PushPullStage` enforces this mostly at compile time with proper types, but `GraphStage` does not. Btw this simple invariant is what allows us to make these interpreters allocation free.